staging/src/kubevirt.io/client-go/api/v1/hyperconvergedconfig.go

```diff
@@ -0,0 +1,39 @@
+package v1
+
+// This code is copied from
+// https://github.com/kubevirt/hyperconverged-cluster-operator/blob/master/pkg/apis/hco/v1beta1/hyperconverged_types.go
+// in order to avoid dependency loops
+
+import (
+	corev1 "k8s.io/api/core/v1"
+)
+
+// HyperConvergedConfig is the schema for pod configuration that is going to
+// be propagated by HCO to its managed operators.
+//
+// +k8s:openapi-gen=true
+type HyperConvergedConfig struct {
```

Comment:
nit: This is actually also for KubeVirt the preferred way of shaping the placement over the customization-patch-way. I would therefore vote for a struct name like `ComponentPlacement` or the like.

Reply:
@tiraboschi was the name HyperConvergedConfig supposed to be literal for all projects? Is there an issue with calling this ComponentPlacement?

Reply:
> @tiraboschi was the name HyperConvergedConfig supposed to be literal for all projects? Is there an issue with calling this ComponentPlacement?

I think we need to use the new [SDK](https://github.com/kubevirt/controller-lifecycle-operator-sdk) for that. Currentlly we merged 3 diffrent implementation into HCO, and then we must also translate each one of them.

Reply:
@nunnatsa you answer confused me somewhat. Are you suggesting that we need to re-architect this PR? Is this SDK required if we wanted to change the name? Would it still be something we needed if we followed suit and used HCO's naming convention?

Reply:
> Are you suggesting that we need to re-architect this PR?

I guess we don't have time for that. But I would do it in the future. 

> Is this SDK required if we wanted to change the name? 

No. This only for HCO to use the same go type for all the ipwrators,and to avoid translation when passong the data. 

Would it still be something we needed if we followed suit and used HCO's naming convention?

No. But it will be easier.

Reply:
The main issue is that this feature has nothing to do with HCO as such. It just appears that HCO wants to use it too (like numerous users which requested an easy way to influence the placement). We export the struct names with our kubevirt.io/client-go and it will also be in the api spec. This can be confusing.

Reply:
@tiraboschi, @rmohr makes a very good point. Once this is merged/released it becomes pretty difficult to change. Is a more product-agnostic naming convention permissible here?

Reply:
We can change it later on. We don't guarantee struct-stability (only api stability). So I have no intent on blocking on that, but I would want to see it at least considered.

Reply:
So I moved HCO to use the sdk version and it was almost immediate change.

The naming there has nothing to do with HCO and I suggest agian to adopt it. the change is minor, because the field names and types are the same, because they also adopted HCO structure:

```golang
// NodePlacement describes node scheduling configuration.
// +k8s:openapi-gen=true
type NodePlacement struct {
	// nodeSelector is the node selector applied to the relevant kind of pods
	// It specifies a map of key-value pairs: for the pod to be eligible to run on a node,
	// the node must have each of the indicated key-value pairs as labels
	// (it can have additional labels as well).
	// See https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#nodeselector
	// +kubebuilder:validation:Optional
	// +optional
	NodeSelector map[string]string `json:"nodeSelector,omitempty"`

	// affinity enables pod affinity/anti-affinity placement expanding the types of constraints
	// that can be expressed with nodeSelector.
	// affinity is going to be applied to the relevant kind of pods in parallel with nodeSelector
	// See https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#affinity-and-anti-affinity
	// +kubebuilder:validation:Optional
	// +optional
	Affinity *corev1.Affinity `json:"affinity,omitempty"`

	// tolerations is a list of tolerations applied to the relevant kind of pods
	// See https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/ for more info.
	// These are additional tolerations other than default ones.
	// +kubebuilder:validation:Optional
	// +optional
	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
}
```
(source: https://github.com/kubevirt/controller-lifecycle-operator-sdk/blob/master/pkg/sdk/api/types.go#L49)

Reply:
The HCO PR: https://github.com/kubevirt/hyperconverged-cluster-operator/pull/801

Reply:
`NodePlacement` sounds better than `HyperConvergedConfig`. :+1:

Reply:
@rmohr - it's more than just naming. I'm talking about the same go struct.

Reply:
As usual, we debated a lot about namings also when we introduced the APIs on HCO side.
See for instance: https://github.com/kubevirt/hyperconverged-cluster-operator/pull/718#issuecomment-672973258

Initially it was exactly called placement also there, but then we decided to make it more generic and call it HyperConvergedConfig to let us inject more values to be propagated to components operator in the future.
So we decided to nest placement under that, then we decided to remove it because NodeSelector/Affinity/Tolerations was declarative enough.

Reply:
The discussion here is of course independent of the naming in HCO itself. Having Hyperconverged here in the struct name simply does not make sense imo. Simone, do I miss something?

Reply:
We named it HyperConvergedConfig on HCO CR to make more evident that it's content is going to be propagated to other components (so hyperconverged) and it's not HCO specific.
HyperConvergedConfig is just a container of various configuration options, NodePlacement is just the first one.

Reply:
Do you suggest to not name it NodePlacement? Do we agree that it does not make sense to introduce a struct with the name HyperConverged in the kubevirt operator? It just deploys kubevirt, there is nothing "hyperconverged" here.
I think I miss what you want to say.

Reply:
We want to use HCO as a proxy for configurations that relevant to all (or most of) the other operators. Right now it's node placement, but this name is limiting onlyu for node placement use cases.

Reply:
I can still not follow. All I care about is that it is not called `HyperConvergedSomething`, since it has nothing to do with hyperconverged in kubevirt. If you feel more comfortable with another name, just propose it.

Reply:
We can also again rename the struct later if we like, but it is obvious that `HyperConverged` is confusing and wrong in this context, therefore I want to avoid it.

Reply:
Not using "HyperConverged" is understanable. But since it was wrapped in other places as "NodePlacement" it's to specific for what it should do in HCO, so we nested it:

```golang
type HyperConvergedConfig struct {
	// NodePlacement describes node scheduling configuration.
	// +optional
	NodePlacement *sdkapi.NodePlacement `json:"nodePlacement,omitempty"`
}
```

Reply:
We can nest it here too, sounds good. Something like `ComponentConfig` could be good enough.